lib/ex_doc/refs.ex

```diff
@@ -100,17 +100,14 @@ defmodule ExDoc.Refs do
   defp fetch_entries(module, result) do
     case result do
       {:docs_v1, _, _, _, module_doc, _, docs} ->
-        module_visibility = visibility(module_doc)
+        module_visibility = visibility({module, module_doc})
 
         for {{kind, name, arity}, _, _, doc, metadata} <- docs do
-          visibility =
-            case {module_visibility, visibility(doc)} do
-              {:hidden, :public} -> :hidden
-              {_, visibility} -> visibility
-            end
+          ref_kind = to_ref_kind(kind)
+          visibility = visibility({module, module_doc}, {ref_kind, name, doc})
```

Comment:
I think this can be simpler. Something like this:

    if module_visibility == :hidden do
      :hidden
    else
      def_visibility(ref_kind, name, doc)
    end

Reply:
I don't think we should have this visibility logic placed in a fetch function. Actually, the if i'm  not mistaken the very same bug that this PR solves is because of this logic being out of the visibility function.

Reply:
You can put it inside the visibility function if you want. My point is that the logic today is more complicated than it needs to be, because you can work with the module_visibility you already got, instead of computing it over again.

Reply:
But removing the `module` gets a similar result, so it looks good to me! :)